src/Microsoft.OpenApi.Hidi/OpenApiService.cs

```diff
@@ -18,123 +21,200 @@
+                var requestUrls = ParseJsonCollectionFile(fileStream, logger);
 
-                foreach (var error in context.Errors)
-                {
-                    errorReport.AppendLine(error.ToString());
-                }
+                logger.LogTrace("Creating predicate based on the paths and Http methods defined in the Postman collection.");
+                predicate = OpenApiFilterService.CreatePredicate(requestUrls: requestUrls, source:document);
 
-                throw new ArgumentException(string.Join(Environment.NewLine, context.Errors.Select(e => e.Message).ToArray()));
+                logger.LogTrace("Creating subset OpenApi document.");
+                document = OpenApiFilterService.CreateFilteredDocument(document, predicate);
             }
-
+            
+            logger.LogTrace("Creating a new file");
             using var outputStream = output?.Create();
-
-            var textWriter = outputStream != null ? new StreamWriter(outputStream) : Console.Out;
+            var textWriter = outputStream != null ? new StreamWriter(outputStream) : Console.Out;            
 
             var settings = new OpenApiWriterSettings()
             {
                 ReferenceInline = inline ? ReferenceInlineSetting.InlineLocalReferences : ReferenceInlineSetting.DoNotInlineReferences
             };
 
-            var openApiFormat = format ?? GetOpenApiFormat(openapi);
+            var openApiFormat = format ?? GetOpenApiFormat(openapi, logger);
             var openApiVersion = version ?? result.OpenApiDiagnostic.SpecificationVersion;
             IOpenApiWriter writer = openApiFormat switch
             {
                 OpenApiFormat.Json => new OpenApiJsonWriter(textWriter, settings),
                 OpenApiFormat.Yaml => new OpenApiYamlWriter(textWriter, settings),
                 _ => throw new ArgumentException("Unknown format"),
             };
+
+            logger.LogTrace("Serializing to OpenApi document using the provided spec version and writer");
+            
+            stopwatch.Start();
             document.Serialize(writer, openApiVersion);
+            stopwatch.Stop();
+
+            logger.LogTrace($"Finished serializing in {stopwatch.ElapsedMilliseconds}ms");
 
             textWriter.Flush();
         }
 
-        private static Stream GetStream(string openapi)
+        private static Stream GetStream(string input, ILogger logger)
         {
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
+
             Stream stream;
-            if (openapi.StartsWith("http"))
+            if (input.StartsWith("http"))
             {
-                var httpClient = new HttpClient(new HttpClientHandler()
+                try
                 {
-                    SslProtocols = System.Security.Authentication.SslProtocols.Tls12,
-                })
+                    var httpClient = new HttpClient(new HttpClientHandler()
```

Comment:
```suggestion
                    using var httpClient = new HttpClient(new HttpClientHandler()
```

and move the client handler to a variable, with a using statement as well

Reply:
Resolving..